tests/luagit2_commit_test.lua

```diff
@@ -0,0 +1,234 @@
+	commit_id = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
+	lunatest.assert_not_nil(commit_id)
+
+	commit = lib.luagit2_commit_lookup(repo,commit_id)
+	lunatest.assert_not_nil(commit)
+
+	committer_signature = lib.luagit2_commit_committer(commit)
+	lunatest.assert_not_nil(committer_signature)
+	lunatest.assert_equal("Satyendra_Kumar_Banjare" , lib.luagit2_get_signature_details(committer_signature,1))
+	lunatest.assert_equal("satyendrabanjare99@gmail.com" , lib.luagit2_get_signature_details(committer_signature,2))
+end
+
+function luagit2_commit_tests.test_luagit2_commit_message()
+	repo = lib.luagit2_repository_open("./Simple-Repository")
+	lunatest.assert_not_nil(repo)
+	-- this is an example commit id for a commit in the repository Simple-Repository
+	commit_id = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
+	lunatest.assert_not_nil(commit_id)
+
+	commit = lib.luagit2_commit_lookup(repo,commit_id)
+	lunatest.assert_not_nil(commit)
+
+	commit_message = lib.luagit2_commit_message(commit)
+	lunatest.assert_not_nil(commit_message)
+	lunatest.assert_string(commit_message)
+end
+
+function luagit2_commit_tests.test_luagit2_commit_message_encoding()
+	repo = lib.luagit2_repository_open("./Simple-Repository")
+	lunatest.assert_not_nil(repo)
+	-- this is an example commit id for a commit in the repository Simple-Repository
+	commit_id = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
+	lunatest.assert_not_nil(commit_id)
+
+	commit = lib.luagit2_commit_lookup(repo,commit_id)
+	lunatest.assert_not_nil(commit)
+
+	commit_message_enc = lib.luagit2_commit_message_encoding(commit)
+	lunatest.assert_nil(commit_message_enc) -- there is no encoding to the message so it should be nil.
+end
+
+function luagit2_commit_tests.test_luagit2_commit_message_raw()
+	repo = lib.luagit2_repository_open("./Simple-Repository")
+	lunatest.assert_not_nil(repo)
+	-- this is an example commit id for a commit in the repository Simple-Repository
+	commit_id = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
+	lunatest.assert_not_nil(commit_id)
+
+	commit = lib.luagit2_commit_lookup(repo,commit_id)
+	lunatest.assert_not_nil(commit)
+
+	commit_message_raw = lib.luagit2_commit_message_raw(commit)
+	lunatest.assert_not_nil(commit_message_raw)
+	lunatest.assert_string(commit_message_raw)
+end
+
+function luagit2_commit_tests.test_luagit2_commit_nth_gen_ancestor()
+	repo = lib.luagit2_repository_open("./Simple-Repository")
+	lunatest.assert_not_nil(repo)
+	-- this is an example commit id for a commit in the repository Simple-Repository
+	commit_id = lib.luagit2_oid_fromstr("bb300684ed4da2e11104a98ff4d825d81787ad3a")
+	lunatest.assert_not_nil(commit_id)
+
+	commit = lib.luagit2_commit_lookup(repo,commit_id)
+	lunatest.assert_not_nil(commit)
+
+	ancestor_commit = lib.luagit2_commit_nth_gen_ancestor(commit,1)
+	lunatest.assert_not_nil(ancestor_commit)
```

Comment:
See comments above about generating the commits for the tests :) 
You need to know it returns the correct commit, not just a value different from null!